Core/AppRuntime/Source/WorkQueue.h

```diff
@@ -19,32 +19,44 @@ namespace Babylon
         template<typename CallableT>
         void Append(CallableT callable)
         {
-            std::scoped_lock lock{m_appendMutex};
-            m_task = m_task.then(m_dispatcher, m_cancelSource, [this, callable = std::move(callable)]() mutable noexcept {
-                try
-                {
-                    callable(m_env.value());
-                }
-                catch (...)
-                {
-                    m_unhandledExceptionHandler(std::current_exception());
-                }
-            });
+            if constexpr (std::is_copy_constructible<CallableT>::value)
```

Comment:
I don't understand this. Why do you need to copy?

Reply:
The callable to be queued needs to be copyable in order for the dispatcher to make a copyable inplace function from it. I didn't dive all the way into why the work queue needs its inplace functions to be copyable; I feel like we've been down that road before, but I don't remember the particulars. What I encountered was that the compiler complained if I tried to put noncopyable callables onto the dispatcher directly, so I worked around it for scenarios where the callable has no copy constructor but does have a move constructor (for example, if the copyable captured a `unique_ptr`).